code/src/main/java/frc/robot/commands/navcommands/TurnToHeading.java

```diff
@@ -63,19 +77,22 @@ public void execute() {
      * @param deadZone the maximum value to be considered zero
      * @return the modified version of input
      */
-    public double deadZones(double input, double deadZone) {
-        if (Math.abs(input) < deadZone) {
-            return 0;
-        }
-        return input;
-    }
 
     /**
      * this command controllers a controller so will run forever!
      */
     @Override
     public boolean isFinished() {
-        return false;
+
+        return _onTargetTicks >= _config.turnToHeading.pidOnTargetTicksGoal && _PidController.atSetpoint();
```

Comment:
Should this be an OR? Ie you're done if you're where you want to be or you've expired the amount of time to get there? Also is this enough time? This assumes you'll get there within 1 sec right?

Reply:
this is not a time based rotation. the time is only used to count how long the bot has been on target. If you just check if you are on target, you will either undershoot or overshoot. the on target ticks check makes sure that the bot has been at the right heading for 5 ticks to ensure that the bot is actually there. The second check is just more redundancy to prevent the command from ending prematurely. If you want it to have a timeout, you can add a withTimeout() call  to ensure the right amount of time.

Reply:
Ok - it might be useful to put a comment that describes that logic as it may not (as in my case) be obvious what its doing.

Reply:
That second `&& pidcontrollee.atSetpoint()` is kind of optional anyways